### ISSUE TITLE
Issue with Website Agent Pulling a Json with a Text Emote

### ISSUE THREAD

**Issue by Kalimeme** [huginn/huginn#2165]
So I am having an issue with a Website Agent looping after pulling a json file. The problem is when one of the posts have a Copy and Paste emote.

I am not sure if there is a way around this.

https://www.reddit.com/r/birdswitharms/new.json This is the json link but here are some picture of what it is doing. I just continues to loop the pull with the event.

It is something to do with the emote because another subreddit has the same problem.
![screenshot_7](https://user-images.githubusercontent.com/15113963/32766225-fadf808a-c8db-11e7-8694-e9ab123faabb.png)
![screenshot_8](https://user-images.githubusercontent.com/15113963/32766226-faee45b6-c8db-11e7-942f-bdf3f8a45af5.png)
![screenshot_9](https://user-images.githubusercontent.com/15113963/32766227-faf7b9b6-c8db-11e7-9245-15025f72ccf0.png)
![screenshot_10](https://user-images.githubusercontent.com/15113963/32766228-fb0130f4-c8db-11e7-8f34-3dbf965df609.png)

**Comment by irfancharania** [comment]
There is a **data.id** field. It looks like it's just plain alphanumeric.
How about using that as the "id"?

**Comment by Kalimeme** [comment]
Well I basically i'm pulling new posts from reddit then using webhooks with json to post back to a discord server as a bot post. So I have no need for the ID, Was looking for the title because if you have ever been on reddit some picture can not really be shown without there title.

**Comment by Kalimeme** [comment]
It also does not really solve the problem of that something is not going right with the code. I'm going to dig into it over the next couple of days, but im more of a hardware guy so wanted to see if anyone had any idea as to what it could be.

**Comment by aserpi** [comment]
I tried with [r/emojipasta](https://www.reddit.com/r/emojipasta/), but i cannot reproduce the issue.

```
{
  "expected_update_period_in_days": "2",
  "url": "https://www.reddit.com/r/emojipasta/new.json",
  "type": "json",
  "mode": "on_change",
  "extract": {
    "id": {
      "path": "data.children[*].data.title"
    },
    "uploader": {
      "path": "data.children[*].data.subreddit"
    },
    "image": {
      "path": "data.children[*].data.url"
    }
  }
}
```
![testagent](https://user-images.githubusercontent.com/15035568/32942432-54430e30-cb89-11e7-9e90-4154b82ed1be.png)
![screenshot-2017-11-17 event 713 - huginn](https://user-images.githubusercontent.com/15035568/32942513-8a003e3a-cb89-11e7-807a-50169b2a6d58.png)

In addition, all emojis are correctly identified and displayed, maybe it is an encoding problem?
Dry run emojipasta with and without forcing the encoding to UTF-8 and see if there is any difference in the log.

**Comment by Kalimeme** [comment]
How would you force the encoding to UTF-8? Also when you have it in did you just try dry running a couple of times? I still have the issue with r/emojipasta, it will just keep making the same events. 
https://i.imgur.com/t47uhqX.png

**Comment by aserpi** [comment]
Yes, no problem whatsoever on my side.
You could try adding the option `"force_encoding":"UTF-8"` to the agent, although I do not think it would change anything: from what I can see, your server supports Unicode up to v5.2 (which is quite strange, considering that v6.0 dates back to 2010) whereas Reddit accepts more recent characters as well. If your server encounters an unknown character, it enters in a loop.

**Comment by Kalimeme** [comment]
Yeah that is strange, I will give it a try. I'm not sure what is causing it. This is a new server that is setup with a the newest version of Ubuntu Server and a fresh install of Huginn.

**Comment by Kalimeme** [comment]
Aserpi: Yeah still having issues with it. When I dry run the both with or without Force, I still have the same problem. I checked my server and it is running UTF-8 my /etc/default/locale has this. https://i.imgur.com/k3XENHD.png
The emotes show up fine when I dry run but when ever the agent is run automatically it generates events  with the ????? errors.

**Comment by dsander** [comment]
@Kalimeme Not sure if it is related, which ruby version and database are you using? I am using ruby 2.4.2 from with PostgreSQL and don't seem to have issues with `emojipasta`

**Comment by somato** [comment]
This is an old issue, so I'm not sure if you're still having trouble with it, but I just had to deal with it myself and thought I'd add my solution.  Using just straight utf8 isn't enough, as most the emoji characeters won't display.  You need to change mysql to use the extended utf8mb4 character set.

After changing the mysql db from utf8 to utf8mb4 (I used https://mathiasbynens.be/notes/mysql-utf8mb4 along with some scripts I found via googling to make it easier), I was still getting ??? for emoji and a looping website agent.  It turns out I also had to tell Huginn to use utf8mb4 via the DATABASE_ENCODING line in the .env file, and that resolved the issue for me.
